### ISSUE TITLE
How should I set "storage.lvm_vg_name" for my server now?

### ISSUE THREAD

**Issue by makarych** [lxc/lxd#2914]
Hello!

After installing new lxd v2.9.1 I cannot apply my presets for lxd server:

```
root@server:~# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial

root@server:~# uname  -r
4.6.7-040607-generic

root@server:~# lxd --version
2.9.1

root@server:~# dpkg -l | grep -iE 'lx(c|d)'                                                                                    
ii  liblxc1                            2.0.7-0ubuntu1~ubuntu16.04.1~ppa1 amd64        Linux Containers userspace tools (library)
ii  libpam-cgfs                        2.0.6-1~ubuntu16.04.1~ppa1        amd64        PAM module for managing cgroups for LXC
ii  lxc-common                         2.0.7-0ubuntu1~ubuntu16.04.1~ppa1 amd64        Linux Containers userspace tools (common tools)
ii  lxc-templates                      2.0.7-0ubuntu1~ubuntu16.04.1~ppa1 amd64        Linux Containers userspace tools (templates)
ii  lxc1                               2.0.7-0ubuntu1~ubuntu16.04.1~ppa1 amd64        Linux Containers userspace tools
ii  lxcfs                              2.0.6-1~ubuntu16.04.1~ppa1        amd64        FUSE based filesystem for LXC
ii  lxd                                2.9.1-0ubuntu3~ubuntu16.04.1~ppa1 amd64        Container hypervisor based on LXC - daemon
ii  lxd-client                         2.9.1-0ubuntu3~ubuntu16.04.1~ppa1 amd64        Container hypervisor based on LXC - client
ii  lxd-tools                          2.9.1-0ubuntu3~ubuntu16.04.1~ppa1 amd64        Container hypervisor based on LXC - extra tools
ii  python3-lxc                        2.0.7-0ubuntu1~ubuntu16.04.1~ppa1 amd64        Linux Containers userspace tools (Python 3.x bindings)

root@server:~# dpkg -l | grep -i 'lvm'
ii  liblvm2app2.2:amd64                2.02.133-1ubuntu10                amd64        LVM2 application library
ii  liblvm2cmd2.02:amd64               2.02.133-1ubuntu10                amd64        LVM2 command library
ii  lvm2

root@server:~# lxc config set storage.lvm_fstype xfs                                                                           
error: Setting the key "storage.lvm_fstype" is deprecated in favor of storage pool configuration.

root@server:~# lxc config set storage.lvm_vg_name "vg0"
error: Setting the key "storage.lvm_vg_name" is deprecated in favor of storage pool configuration.
```
Previously used version lxd-2.6.2 works fine.

**Comment by stgraber** [comment]
With LXD 2.9 you need to create a LVM backed storage pool with the volume.block.filesystem property set to xfs.

@brauner can you provide with the right commands? I haven't played with the LVM backend yet.

**Comment by brauner** [comment]
Yes, I was just about to answer.
So the lvm implementation will allow you to use an existing block device but currently not an existing vg. This will be changed in the next LXD releases. The easiest way to create a lvm storage pool right now is to use the storage client tool via:

```
lxc storage create <pool-name-you-want> source=/dev/sdX
```

LXD should then create a new physical volume, volume group and thinpool for you.

**Comment by stgraber** [comment]
@brauner oh, I didn't realize we didn't support specifying an existing VG as the source for a new LVM storage pool. We need to get that fixed as it's a clear regression from 2.8...

**Comment by brauner** [comment]
Fixed in #2921.
